src/main/java/com/devcycle/sdk/server/local/managers/EventQueueManager.java

```diff
@@ -0,0 +1,181 @@
+package com.devcycle.sdk.server.local.managers;
+
+import com.devcycle.sdk.server.common.api.IDVCApi;
+import com.devcycle.sdk.server.common.model.*;
+import com.devcycle.sdk.server.local.api.DVCLocalEventsApiClient;
+import com.devcycle.sdk.server.local.bucketing.LocalBucketing;
+import com.devcycle.sdk.server.local.model.*;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import retrofit2.Call;
+import retrofit2.Response;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class EventQueueManager {
+
+    private LocalBucketing localBucketing;
+    private final String serverKey;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private IDVCApi eventsApiClient;
+    private int eventFlushIntervalMS;
+    private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private boolean isFlushingEvents = false;
+    private int flushEventQueueSize;
+    private int maxEventQueueSize;
+
+    public EventQueueManager(String serverKey, LocalBucketing localBucketing, DVCLocalOptions options) throws Exception {
+        this.localBucketing = localBucketing;
+        this.serverKey = serverKey;
+        eventFlushIntervalMS = options.getEventFlushIntervalMS();
+        flushEventQueueSize = options.getFlushEventQueueSize();
+        maxEventQueueSize = options.getMaxEventQueueSize();
+
+        if (flushEventQueueSize >= maxEventQueueSize) {
+            throw new Exception("flushEventQueueSize: " + flushEventQueueSize + " must be larger than maxEventQueueSize: " + maxEventQueueSize);
+        } else if (flushEventQueueSize > 20000 || maxEventQueueSize > 20000 ) {
+            throw new Exception("flushEventQueueSize: " + flushEventQueueSize + " or maxEventQueueSize: " + maxEventQueueSize + " must be smaller than 20,000");
+        }
```

Comment:
Can you also check it against the chunk size: 
<img width="867" alt="image" src="https://user-images.githubusercontent.com/1219069/191100510-adf401b2-e8ba-4ed5-8451-4be38fa1eb61.png">